### PR TITLE
Fix #106 #118: Show/Search and Delete branches and tags from dropdown

### LINKED ISSUE #106: browse and search existing tags
#### Feature request:
This is a decomposition of issue #40. The functionality required here should allow the user to:

1. browse all existing tags

2. search for a certain tag

### PR DESCRIPTION
**Pull request by jamshale**
Implemented a feature on top of existing codebase to separate tags and branches. Made the search input field work and refactored both the design and code. 

Branches:

image

### CODE REVIEW

File: app/components/header/header.component.html [hunk 1]
```diff
@@ -12,19 +12,29 @@
...
+                <span class="select-menu-title"> Switch branches/tags</span>
+              </div>
+              <div class="input-group menuitem">
+                <div class="select-menu-text-filter" style="padding:10px">
+                    <span class="input-group-btn">
+                    <input type="text" id="branchName" class="form-control" style="width:320px" placeholder="Find or create branch" oninput="displayBranchesTags()">
+                    <button class="btn btn-default" type="button" onclick="createBranch()">OK</button>
+                    </span>
+                </div>
+                <button class="btn btn-default" style="width:45%; margin: 0 1% 5px 4%" (click)=displayBranches()>Branches</button>
```
Comment (zeshili) [inline app/components/header/header.component.html]: The CSS style should be implemented from a linked component style sheet (i.e. header.component.css) or inline in header.component.ts

File: stylesheets/styles.css [hunk 2]
```diff
@@ -886,6 +886,11 @@ div.wiki-header button.close-button, div.wiki-header button.open-button {
   margin-top: -120px;
 }
 
+#branchesTab .list-group-item, #tagsTab .list-group-item{
+  width:300px;
+  margin: px;
+} 
+
```
Comment (zeshili) [inline stylesheets/styles.css:918]: These styles also seem like they could be in a header.component.css file instead of a global stylesheet

**Review by SrishtiSinghal — APPROVED** [review]
-The code is clean and easy to understand.
-The code could do with some more commenting, but the logic is straight forward and understandable.
-The code works as expected on my macOS version 10.13.6 
-I can search for branches and tags however it doesn't let me create a new branch. After chatting with Jamie, we concluded that we will implement that feature in the next sprint. 
-Other than that everything looks solid, good job Jamie!

**Review by AhmadAlsaif — APPROVED** [review]
Does the code work?

- Yes, it works as intended!

Did I test it locally?

- Yes tested it on my MacOS MoJave

Does it perform its intended function, the logic is correct etc.

- logic is correct and works as intended 

Is the code well documented?

- yes, adequate comments were added. the code makes sense to me

Does it have correct linting/formatting?

- Yes

Is there any redundant or duplicate code? Any hard-coded values?

- Nothing I could find

Can any of the code be replaced with library functions?

- Can't see any

Are good names used for methods, variables, etc.? Do they follow naming conventions?

- Yes

**Pull request merged**